finetune/target_models/deployment_model.py

```diff
@@ -0,0 +1,340 @@
+            original_model.multi_label = self.multi_label
+            self._initialize = original_model._initialize
+        elif isinstance(original_model.input_pipeline, ComparisonPipeline):
+            self.task = TaskMode.COMPARISON
+        elif isinstance(original_model.input_pipeline, BasePipeline):
+            self.task = TaskMode.CLASSIFICATION
+        elif isinstance(original_model.input_pipeline, AssociationPipeline):
+            self.task = TaskMode.ASSOCIATION
+        else:
+            raise FinetuneError("Invalid pipeline in loaded file.")
+
+        self.input_pipeline.task = self.task
+
+    def _get_estimator(self, portion):
+        assert portion in ['featurizer', 'target'], "Can only split model into featurizer and target."
+        conf, distribute_strategy, config = self._get_estimator_config()
+
+        fn = get_separate_model_fns(
+            target_model_fn=self._target_model if portion == 'target' else None, 
+            predict_op=self._predict_op,
+            predict_proba_op=self._predict_proba_op,
+            build_target_model=self.input_pipeline.target_dim is not None,
+            encoder=self.input_pipeline.text_encoder,
+            target_dim=self.input_pipeline.target_dim if portion == 'target' else None,
+            label_encoder=self.input_pipeline.label_encoder if portion == 'target' else None,
+            saver=self.saver,
+            portion=portion,
+            build_attn=not isinstance(self.input_pipeline, ComparisonPipeline)
+        )
+
+        if portion == 'featurizer':
+            estimator = tf.estimator.Estimator(
+                model_dir=self.estimator_dir,
+                model_fn=fn,
+                config=config,
+                params=self.config
+            )
+
+        else:
+            estimator = tf.estimator.Estimator(
+                model_dir=self.estimator_dir,
+                model_fn=fn,
+                config=config,
+                params=self.config
+            )
+
+        if hasattr(self,'predict_hooks') and portion == 'featurizer':
+            for hook in self.predict_hooks:
+                hook.need_to_refresh = True
+        elif not hasattr(self,'predict_hooks'):
+            feat_hook = InitializeHook(self.saver, model_portion='featurizer')
+            target_hook = InitializeHook(self.saver, model_portion='target')
+            predict_hook = namedtuple('InitializeHook', 'feat_hook target_hook')
+            self.predict_hooks = predict_hook(feat_hook, target_hook)
+        return estimator
+
+    def _get_input_pipeline(self):
+        return self.input_pipeline
+
+    def featurize(self, X):
+        """
+        Embeds inputs in learned feature space. Can be called before or after calling :meth:`finetune`.
+
+        :param X: list or array of text to embed.
+        :returns: np.array of features of shape (n_examples, embedding_size).
+        """
+        raise NotImplementedError
+
+    def get_input_fn(self, gen):
```

Comment:
We're mixing and matching public / private naming conventions here.  Seems like these should all obey the same conventions as _get_input_pipeline()`.